### ISSUE TITLE
Automatically Create a new Location during ADIF Import

### ISSUE THREAD

**Issue by Loki1737** [wavelog/wavelog#788]
**What Feature is Missing?**
An ADIF File provides all the information needed to create a station location. Nevertheless do I habe to create manually the station location, before I could import the ADIF file. 

**Why is This Feature Important?**
It would be a great help, if the Location is automatically created based on the information in the file, this reduces manual work and improves the automation. 

**Is This Feature Personal or Beneficial to Others as Well?**
I believe it is helpful for all people who are working at a couple of different locations. So especially for the POTA & SOTA Community. 

Due to the experted impact regarding the implementation work, I understand if this will have to wait some time in the backlog. 

BR,
DO5LE
Lucas

**Comment by DocViper666** [comment]
Full support from my side as a SOTA activator. 
It is just impossible to add hundreds of locations (for each unique summit I have activated) manually.
So I can see an option to split (group-by like) and imported adif by certain conditions (MY_SOTA, MY_POTA, MY_GRIDSQUARE+STATION_CALLSIGN) and list the new locations to be created.

Or provide an API for creation of locations (or locations with their qsos). I would then created an importer tool to push the locations and qso from one or more adif files. I think that would be actually preferable? 

73 Joe, OE5JFE

**Comment by remy56k** [comment]
I agree, in /P (or /M), it's weird to create a station location for each different grids. 

I have tried to keep only one location for my /P (because I always setup the correct current grid in WSJT and each grids are stored/sent in the QSO), but all Wavelog entries are referenced to a fixed single grid defined in the profile, not the imported QSO grids. And some ADI files can have many grids (JT8 roadtrip :smile: ).

Maybe it's a better option to rely on the QSO grid and not on the fixed one ? It will be more permissive and accurate, and let people to manage one CALLSIGN/P with many grids instead create hundred of "Station Locations", and switch in WaveLog-Gate. It could be an option in "Station Locations" : no set the grid = rely on the QSO grid.

Thanks, 
73

**Comment by JayToTheAy** [comment]
It could maybe be done as a checkbox on the ADIF upload page? If a user checks it, whatever data is present for location (WWFF, POTA, IOTA, Grid, eQSL Nickname, etc.) on the **first** QSO record in the log is used to create a new location and this new location is used for that upload. Something like this?

<img width="1251" height="91" alt="Image" src="https://github.com/user-attachments/assets/6d89d4c2-42e1-469e-8769-19b8396a1172" />

I agree that picking which location to use in a potentially contradictory log is a bit arbitrary, but I think first makes it something that a user can easily check by just opening the file, and as long as it's clearly communicated to a user, it would do a lot to simplify the onboarding process when you're coming from a logger that doesn't care about location.

**Comment by roleohibachi** [comment]
Another user story for the pile: I'm new to contest-style logging. Wavelog has been great, in general. I imported each of my ADIFs from each of my SOTA activations, then sync'ed to LotW (cool!).

Then, I realized wavelog was discarding my logged location and replacing it with the default station location/gridsquare. It even **deleted** my own SOTA summit references since the default station location didn't have a SOTA ref. Not good.

I'm struggling to understand the scenario where deleting valid log data during import would be useful behavior. Is it a contest thing, to unify logs collected from various applications into one coherent station location? 

A less dangerous option might be to create a new temporary Station for every import. Then, on the station management panel, offer the ability to merge the import-station with a previously configured station if that is desired. A fancy solution could even detect when stations were exact duplicates.

**Comment by cwhelchel** [comment]
As someone who just tried to import my master log (30k qsos of home QTH plus ~100 POTA/SOTA activations) something like this feature would be much appreciated. 

A flag on import like TQSL, 'Use station info from log' would be very useful as my logs are as accurate and I'd like to carry the MY_GRIDSQUARE, MY_SIG, and MY_SIG_INFO info into wavelog.

**Comment by DocViper666** [comment]
Or what would help already is a method to list all stations _api/list_all_stations_

This would allow to iterate e.g. over the respective SOTA, POTA or other criteria (like MY_GRIDSQUARE) and find the suitable _station_profile_id_ and then upload using the _api/qso/_ API

**Comment by int2001** [comment]
it's already part of station_info endpoint - the endpoint returns all those informations.

see: https://github.com/int2001/waverover

**Comment by int2001** [comment]
going to close this topic, because it's solved with waverover (see above) for every OS.
in near future this wont be part of core-wavelog. Reason: QSOs need to be staged "intermediate" for sorting / deriving Ref, etc.
